src/ProcessedTweetList.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * ProcessedTweetList - Stores tweets/dms by room_id so we don't accidentally
+ * repeat any messages. Please note that the cache and slice size values ar
+ * per room, not the total size of all the rooms.
+ *
+ * @class
+ * @param  {number} cacheSize How many messages to store before cleanup occurs
+ * @param  {number} sliceSize On cleanup, how many messages will be kept
+ */
+var ProcessedTweetList = function(cacheSize,sliceSize){
+    this._internalArray = {};
+    this._cacheSize = cacheSize || 128;
+    this._sliceSize = sliceSize || 16;
```

Comment:
Variable names aren't the most informative. This isn't a generic class, so don't use generic names. You could transplant this into another class and have it make complete sense. I appreciate you've commented what these things do, which is great, but I think you can go further here:
 - `_internalArray` - It's not an array. It's a map of `room_id => Array<TweetID>`. `_roomIdToTweetIds` would be clearer.
 - `_cacheSize` - Cache of what? Your comment says "How many messages to store before cleanup occurs" - so why not something like `maxStoredTweets`?
 - `_sliceSize` - Slice of what? Your comment says "On cleanup, how many messages will be kept" - so why not something like `minStoredTweets`?